contracts/ethereum/src/PaymentRegistry.sol

```diff
@@ -91,7 +96,48 @@ contract PaymentRegistry is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             StarknetEscrowClaimPaymentSelector,
             payload);
 
-        emit ClaimPayment(transferInfo);
+        emit ClaimPayment(orderId, destAddress, amount, Chain.Starknet);
+    }
+
+    function claimPaymentBatch(
+        uint256[] calldata orderIds,
+        uint256[] calldata destAddresses, 
+        uint256[] calldata amounts
+    ) external payable onlyOwnerOrMM() {
+        require(orderIds.length == destAddresses.length, "Invalid lengths.");
+        require(orderIds.length == amounts.length, "Invalid lengths.");
+
+        uint256[] memory payload = new uint256[](5 * orderIds.length + 1);
+
+        payload[0] = orderIds.length;
+        
+        for (uint32 idx = 0; idx < orderIds.length; idx++) {
+            uint256 orderId = orderIds[idx];
+            uint256 destAddress = destAddresses[idx];
+            uint256 amount = amounts[idx];
+
+            _claimPaymentStarknet(orderId, destAddress, amount);
+
+            uint32 base_idx = 1 + 5 * idx;
+            payload[base_idx] = uint128(orderId); // low
+            payload[base_idx + 1] = uint128(orderId >> 128); // high
+            payload[base_idx + 2] = destAddress;
+            payload[base_idx + 3] = uint128(amount); // low
+            payload[base_idx + 4] = uint128(amount >> 128); // high
+        }
+        
+        _snMessaging.sendMessageToL2{value: msg.value}(
+            StarknetEscrowAddress,
+            StarknetEscrowClaimPaymentBatchSelector,
+            payload);
+
+        emit ClaimPaymentBatch(orderIds, destAddresses, amounts, Chain.Starknet);
+    }
+
+    function _claimPaymentStarknet(uint256 orderId, uint256 destAddress, uint256 amount) internal view {
```

Comment:
rename this function to a more descriptive name

Reply:
Renamed on 218253c54eaed2c51c1ab7999cbe2807514e7dd0